### ISSUE TITLE
NGram and Overlap score bug

### ISSUE THREAD

**Issue by obrenckle** [vickumar1981/stringdistance#42]
While two identical strings get a score of 1, in every other cases both NGram and Overlap return a score where 0 = most similar and 1 = completely different:

```scala
NGram.score("karolin", "karolin", 1)  // 1.0
NGram.score("karolin", "karolyn", 1)  // 0.143
NGram.score("karolin", "kathrin", 1)  // 0.286
NGram.score("karolin", "bcdefgh", 1)  // 1.0

Overlap.score("karolin", "karolin", 1)  // 1.0
Overlap.score("karolin", "karalin", 1)  // 0.143
Overlap.score("karolin", "kathrin", 1)  // 0.286
Overlap.score("karolin", "bcdefgh", 1)  // 1.0
```

I suspect that's because interfaces.NGramTokenizer.foldNGram expects its fourth argument to be a function taking both token sequences and the *number of common tokens* in both, while impl.NGramImpl.nGram and impl.OverlapImpl.overlap pass a function with both sequences and the *distance* between them.

**Comment by vickumar1981** [comment]
:+1:  will fix and release a new version this weekend.  thanks, and you are correct about the cause of the bug as well, which makes it easy to fix.

**Comment by vickumar1981** [comment]
@lekeeh   I just pushed a `1.1.4` build to sonatype, and it should mirror to maven central in a few hours or so.  Also, the `1.1.4-SNAPSHOT` should have the changes included as well. (I botched the 1.1.3 build, b/c I forgot that I changed gpg keys).

You were right, and I feel dumb - in most of the cases, distance and score, are inversely proportional, but for nGram, the more nGrams in common, the more alike the two strings are, and the higher the strings are.

Thanks for filing the bug.  If you can confirm that you see the correct behavior in `1.1.4` either from the release when it's available, or from the snapshot, I can close this issue out.

**Comment by obrenckle** [comment]
It's the kind of bug that can easily pass under the radar. The behavior in 1.1.4 is correct, you can close the issue. 
Thanks !

**Comment by vickumar1981** [comment]
Fixed in v1.1.4
